### PR TITLE
Support for FQDNs in Network and NAT Rules for Azure Firewall

### PR DESCRIPTION
**Pull request by tejasshah7**
## Description

Support for FQDNs in Network and NAT Rules for Azure Firewall

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

**Comment by number213** [comment]
Depends on https://github.com/Azure/azure-powershell/pull/10386

**Comment by wyunchi-ms** [comment]
Hi @tejasshah7 , why do you want to merge into master directly? Should not we merge into network-october first?

**Comment by wyunchi-ms** [comment]
> Depends on #10386

Hi @number213 what does depends on #10386 mean? Do this PR will be simpler after merge #10386?
Thanks!

**Comment by number213** [comment]
> Hi @number213 what does depends on #10386 mean? Do this PR will be simpler after merge #10386?

Hi @wyunchi-ms. Yes, there are a bunch of PRs that didn't make into #10386, so they have to go after it. And once #10386 is merged their diff would be much smaller

File: src/Network/Network.Test/ScenarioTests/AzureFirewallTests.cs [hunk 1]
```diff
@@ -34,31 +34,31 @@ public void TestAzureFirewallCRUD()
...
+        [Fact(Skip = "skip")]
         [Trait(Category.AcceptanceType, Category.CheckIn)]
         [Trait(Category.Owner, NrpTeamAlias.azurefirewall)]
         public void TestAzureFirewallAllocateAndDeallocate()
         {
             TestRunner.RunTestScript("Test-AzureFirewallAllocateAndDeallocate");
         }
 
-        [Fact]
+        [Fact(Skip = "skip")]
```
Comment (wyunchi-ms) [inline src/Network/Network.Test/ScenarioTests/AzureFirewallTests.cs]: Why do you want to skip these test cases?

Comment (tejasshah7) [reply src/Network/Network.Test/ScenarioTests/AzureFirewallTests.cs]: That was just for testing. Have reverted this change.

**Pull request merged**